Source/RxSwift/nwise.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  nwise.swift
+//  RxSwiftExt
+//
+//  Created by Zsolt Váradi on 2017. 12. 09..
+//  Copyright © 2017. RxSwiftCommunity. All rights reserved.
+//
+
+import RxSwift
+
+extension ObservableType {
+    /**
+     Groups the elements of the source observable into arrays of N consecutive elements.
+
+     The resulting observable:
+     - does not emit anything until the source emits at least N elements;
+     - emits an array for every element after that;
+     - and forwards any error or completed event.
+
+     For example:
+
+          --(1)--(2)--(3)-------(4)-------(5)------->
+           |
+           | nwise(3)
+           v
+          ------------([1,2,3])-([2,3,4])-([3,4,5])->
+
+     Caveat: n = 0 causes the operator to collapse into: `.map { _ in [] }`.
+
+     - parameter n: size of the groups
+    */
+    public func nwise(_ n: UInt) -> Observable<[E]> {
+        return self
```

Comment:
I understand the use for Uint (non-negative), but I think having Uint and then performing an Int conversion is a bit unneeded and is a bit of boilerplate.

I would just have make it an Int and assert the input.

Two options that come to mind
1. `assert(n > 0, "n must be positive")`, or guard against it and emit an error (might even make sense to assert  `> 1`)
2. use abs(n), but this might confuse a developer.

Reply:
I'll change this to an `assert` then.